src/main/kotlin/lotto/domain/Money.kt

```diff
@@ -0,0 +1,42 @@
+package lotto.domain
+
+import java.math.BigDecimal
+import java.math.RoundingMode
+
+class Money(initMoney: Int) {
+    var spentMoney: Long = 0L
+        private set
+    var currentMoney: Long = initMoney.toLong()
+        private set
+
+    init {
+        checkValidateMoney(initMoney)
+    }
+
+    fun spendMoney(money: Int) {
+        checkEnoughMoney(money)
+        currentMoney -= money
+        spentMoney += money
+    }
+
+    fun spendAllMoney() {
+        spentMoney += currentMoney
+        currentMoney -= currentMoney
+    }
+
+    fun calculateRateOfReturn(winningMoney: Long): String {
+        return BigDecimal(winningMoney)
+            .divide(spentMoney.toBigDecimal(), 2, RoundingMode.FLOOR)
+            .toString()
+    }
+
+    private fun checkEnoughMoney(spendMoney: Int) {
+        if (spendMoney > currentMoney) throw IllegalStateException(
+            "사고자 하는 수량이 현재 가진 돈보다 많습니다.[현재 돈 :$currentMoney 쓰려는 돈: $spendMoney]"
+        )
```

Comment:
`check()`는 값을 만족하지 않으면 `IllegalStateException`을 발생시킵니다.

```suggestion
        check(spendMoney <= currentMoney) {
            "사고자 하는 수량이 현재 가진 돈보다 많습니다.[현재 돈 :$currentMoney 쓰려는 돈: $spendMoney]"
        }
```